pkg/kubelet/kubelet.go

```diff
@@ -1494,39 +1497,15 @@ func (kl *Kubelet) Run(updates <-chan kubetypes.PodUpdate) {
 // the most accurate information possible about an error situation to aid debugging.
 // Callers should not throw an event if this operation returns an error.
 func (kl *Kubelet) syncPod(o syncPodOptions) error {
+	klog.Infof("DEBUG: syncPod %s/%s(%s) enter", o.pod.Namespace, o.pod.Name, o.pod.UID)
+	defer klog.Infof("DEBUG: syncPod %s/%s(%s) exit", o.pod.Namespace, o.pod.Name, o.pod.UID)
+
 	// pull out the required options
 	pod := o.pod
 	mirrorPod := o.mirrorPod
 	podStatus := o.podStatus
 	updateType := o.updateType
 
-	// if we want to kill a pod, do it now!
```

Comment:
Once a pod transitions from "syncing" to "terminating" the pod worker stops calling syncPod and starts calling syncTerminatingPod.  To cause this transition:

1. Delete the pod on apiserver (config change leads to deletion timestamp)
2. Transition restart never pod to succeeded or failed
3. Invoke UpdatePod{UpdateType: Kill} (or use the killPodNow wrapper) for eviction, preemption, and graceful shutdown

This is a one way transition.

Soft admission failures simply call killPod in syncPod, since the next sync might start the container again (soft admission failure is not a termination, it's temporary stop).

Reply:
Will be in the next push.